examples/geostrophic_adjustment.jl

```diff
@@ -0,0 +1,132 @@
+
+# ## A Geostrophc adjustment initial value problem
+#
+# We pose a geostrophic adjustment problem that consists of a Gaussian geostrophic
+# velocity and height field,
+
+Gaussian(x, L) = exp(-x^2 / 2L^2)
+
+U = 0.1 # geostrophic velocity
+L = grid.Lx / 40 # Gaussian width
+x₀ = grid.Lx / 4 # Gaussian center
+
+vᵍ(x, y, z) = - U * (x - x₀) / L * Gaussian(x - x₀, L)
+
+g = model.free_surface.gravitational_acceleration
+
+a = coriolis.f * U * L / g # free surface amplitude
+
+ηᵍ(x, y, z) = a * Gaussian(x - x₀, L)
+
+# We use an initial height field that's twice the geostrophic solution,
+# thus superimposing a geostrophic and ageostrophic component in the free
+# surface displacement field,
+
+ηⁱ(x, y, z) = 2 * ηᵍ(x, y, z)
+
+# We set the initial condition to ``vᵍ`` and ``ηⁱ``,
+
+set!(model, v=vᵍ, η=ηⁱ)
+
+# ## Running a `Simulation`
+#
+# We pick a time-step that resolves the surface dynamics,
+
+gravity_wave_speed = sqrt(g * grid.Lz) # hydrostatic (shallow water) gravity wave speed
+
+wave_propagation_time_scale = model.grid.Δx / gravity_wave_speed
+
+simulation = Simulation(model, Δt = 0.1 * wave_propagation_time_scale, stop_iteration = 1000)
+
+# ## Output
+#
+# We output the velocity field and free surface displacement,
+
+output_fields = merge(model.velocities, (η=model.free_surface.η,))
+
+using Oceananigans.OutputWriters: JLD2OutputWriter, IterationInterval
+
+simulation.output_writers[:fields] = JLD2OutputWriter(model, output_fields,
+                                                      schedule = IterationInterval(10),
+                                                      prefix = "geostrophic_adjustment",
+                                                      force = true)
+
+run!(simulation)
+
+# ## Visualizing the results
+
+using JLD2, Plots, Printf, Oceananigans.Grids
+using Oceananigans.Utils: hours
+
+xη = xw = xv = xnodes(model.free_surface.η)
+xu = xnodes(model.velocities.u)
+
+file = jldopen(simulation.output_writers[:fields].filepath)
+
+iterations = parse.(Int, keys(file["timeseries/t"]))
+
+anim = @animate for (i, iter) in enumerate(iterations)
+
+    u = file["timeseries/u/$iter"][:, 1, 1]
+    v = file["timeseries/v/$iter"][:, 1, 1]
+    η = file["timeseries/η/$iter"][:, 1, 1]
+    t = file["timeseries/t/$iter"]
+
+    titlestr = @sprintf("Geostrophic adjustment at t = %.1f hours", t / hours)
+
+    v_plot = plot(xv / kilometers, v, linewidth = 2, title = titlestr,
+                  label = "", xlabel = "x (km)", ylabel = "v (m s⁻¹)", ylims = (-U, U))
+                  
+    u_plot = plot(xu / kilometers, u, linewidth = 2,
+                  label = "", xlabel = "x (km)", ylabel = "u (m s⁻¹)")
+
+    η_plot = plot(xη / kilometers, η, linewidth = 2,
+                  label = "", xlabel = "x (km)", ylabel = "η (m)", ylims = (-a/10, 2a))
+
+    plot(v_plot, u_plot, η_plot, layout = (3, 1), size = (800, 600))
+end
+
+close(file)
+
+gif(anim, "geostrophic_adjustment.gif", fps = 15) # hide
```

Comment:
change to `mp4`; see rest of examples :)

Reply:
ah true, that was just a placeholder so I could have a gif-driven workflow...